drivers/dma/dma_intel_adsp_hda.c

```diff
@@ -250,6 +250,25 @@ int intel_adsp_hda_dma_start(const struct device *dev, uint32_t channel)
 
 	__ASSERT(channel < cfg->dma_channels, "Channel does not exist");
 
+#if CONFIG_PM_DEVICE
+	bool first_use = false;
+	enum pm_device_state state;
+
+	/* If the device is used for the first time, we need to let the power domain know that
+	 * we want to use it.
+	 */
+	if (pm_device_state_get(dev, &state) != -ENOSYS) {
+		first_use = state != PM_DEVICE_STATE_ACTIVE;
+		if (first_use) {
+			int ret = pm_device_runtime_get(dev);
+
+			if (ret < 0) {
+				return ret;
+			}
+		}
+	}
+#endif
```

Comment:
What is the purpose of this ???? This function ends up calling `pm_device_runtime_get` anyways ...
It seems that you can completely remove this logic and just call `pm_device_runtime_get` at the beginning. One more thing, `pm_device_runtime_get` is a nop operation when building without `CONFIG_PM_DEVICE_RUNTIME` you don't need this `#ifdef`.

Reply:
It's explained in the comment. Power domain need to be power up first, before the device. I know it's ugly but I didn't come up with anything better. And it's not true that `pm_device_runtime_get` it's always called. We want to incremet usage only when new channel is enabled.

Reply:
DMA drivers allow to call start and stop multiple times for the same instance and channel. Therefore, apart from the first use of the **put**/**get** calls are after the status checks `intel_adsp_hda_is_enabled`.